integration_test/src/test/java/com/newrelic/telemetry/SpanApiIntegrationTest.java

```diff
@@ -0,0 +1,137 @@
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+@Testcontainers
+class SpanApiIntegrationTest {
+
+  private static final int SERVICE_PORT = 1080 + new Random().nextInt(900);
+  private static String containerIpAddress;
+  private static MockServerClient mockServerClient;
+  private static final GenericContainer container =
+      new GenericContainer("jamesdbloom/mockserver:mockserver-5.5.1")
+          .withExposedPorts(SERVICE_PORT);
+  private SpanBatchSender spanBatchSender;
+
+  @BeforeAll
+  static void beforeClass() {
+    container.setPortBindings(singletonList(SERVICE_PORT + ":1080"));
+    container.setWaitStrategy(new WaitAllStrategy());
+    container.setStartupCheckStrategy(
+        new MinimumDurationRunningStartupCheckStrategy(Duration.of(10, SECONDS)));
+    container.start();
+    containerIpAddress = container.getContainerIpAddress();
+    mockServerClient = new MockServerClient(containerIpAddress, SERVICE_PORT);
+  }
+
+  @BeforeEach
+  void setUp() throws Exception {
+    mockServerClient.reset();
+    spanBatchSender =
+        SpanBatchSender.builder()
+            .apiKey("fakeKey")
+            .httpPoster(new OkHttpPoster(Duration.ofMillis(1500)))
+            .attributesJson(new AttributesGson(new Gson()))
+            .uriOverride(URI.create("http://" + containerIpAddress + ":" + SERVICE_PORT))
+            .enableAuditLogging()
+            .build();
+  }
+
+  @Test
+  @DisplayName("Low Level SDK can send a span batch and returns a successful response")
+  void testSuccessfulSpanSend() throws Exception {
+    SpanPayload expectedPayload =
+        new SpanPayload(
+            ImmutableMap.of("trace.id", "123456", "attributes", singletonMap("key1", "val1")),
+            singletonList(
+                ImmutableMap.<String, Object>builder()
+                    .put("id", "6f377f46-3f10-4e8e-8390-7717044cdbe9")
+                    .put("trace.id", "123456")
+                    .put("timestamp", 55555)
+                    .put(
+                        "attributes",
+                        ImmutableMap.of(
+                            "duration.ms",
+                            60,
+                            "service.name",
+                            "Span Test Service",
+                            "name",
+                            "spanTest"))
+                    .build()));
+
+    mockServerClient
+        .when(
+            new HttpRequest()
+                .withMethod("POST")
+                .withPath("/trace/v1")
+                .withBody(
+                    json(
+                        new SpanPayload[] {expectedPayload},
+                        MediaType.JSON_UTF_8,
+                        MatchType.ONLY_MATCHING_FIELDS))
+                .withHeader("User-Agent", "NewRelic-Java-TelemetrySDK/.*")
+                .withHeader("Content-Type", "application/json; charset=utf-8")
+                .withHeader("Content-Length", ".*"))
```

Comment:
Is this helpful, or can we omit the wildcard length check?

Reply:
I think this is a good thing to check...it's well-behaved to provide a content length header.